lib/topological_inventory/sync/inventory_upload/payload.rb

```diff
@@ -193,6 +193,39 @@ def sources_api_client(tenant = nil)
 
           SourcesApiClient::DefaultApi.new(api_client)
         end
+
+        def topology_inventory_api_client(tenant = nil)
+          api_client = TopologicalInventoryApiClient::ApiClient.new
+
+          if tenant
+            x_rh_identity = Base64.strict_encode64(JSON.dump("identity" => {"account_number" => tenant}))
+            api_client.default_headers.merge!("x-rh-identity" => x_rh_identity)
+          end
+
+          TopologicalInventoryApiClient::DefaultApi.new(api_client)
+        end
+
+        def find_source_in_topology_inventory(source_uid)
+          api_client ||= topology_inventory_api_client(account)
+          meta_data = api_client.list_sources(:filter => {:uid => source_uid }).meta
+          meta_data.count.positive?
+        end
+
+        TIMEOUT_COUNT = 30
```

Comment:
I would prefer `30.seconds` instead of `30`. 

Anyway, @agrare what do you think about this limit? 30 seconds wait limit seems too low for me, but kafka settings `session_timeout` is 60 seconds, so when we choose more, code (at least with unpacked data) should be running in thread.